snoop/data/analyzers/thumbnails.py

```diff
@@ -342,18 +329,26 @@ def call_thumbnails_service(data, size):
         """
 
     url = settings.SNOOP_THUMBNAIL_URL + f'preview/{size}x{size}'
-    log.info(url)
+    with blob.open() as data:
+        payload = {'file': data}
+        resp = requests.post(url, files=payload)
 
-    resp = requests.post(url, files={'file': data})
     log.info(resp.status_code)
 
-    if resp.status_code == 500:
-        print(resp.text)
-        raise SnoopTaskBroken('thumbnail service returned http 500', 'thumbnail_http_500')
+    if resp.status_code == 404 or resp.status_code == 500:
+        for _ in range(5):
+            time.sleep(random.randint(20, 45))
+            with blob.open() as data:
+                payload['file'] = data
+                resp = requests.post(url, files=payload)
+            if resp.status_code == 200:
+                break
 
-    if (resp.status_code != 200
+    if resp.status_code == 500:
+        raise SnoopTaskBroken(resp.text, 'thumbnail_http_500')
+    elif (resp.status_code != 200
             or resp.headers['Content-Type'] != 'image/jpeg'):
-        raise RuntimeError(f"Unexpected response from thumbnails-service: {resp}")
+        raise RuntimeError(f"Unexpected response from thumbnails-service: {resp} {resp.text}")
```

Comment:
Let's merge these code paths and only raise SnoopTaskBroken (and never RuntimeError), that's either `thumbnail_http_500` or `thumbnail_http_404` or `thumbnail_unknown_error`

We never want this task to halt processing on some given document, and if we raise RuntimeError that's what happens -- processing  is completely halted. Thumbnails are not essential, and we want to keep working even if they're all broken.